examples/phi3/phi3_vision.py

```diff
@@ -0,0 +1,378 @@
+        genai_config = json.load(f)
+    if args.target == "cuda":
+        genai_config["model"]["decoder"]["session_options"]["provider_options"] = [{"cuda": {}}]
+    else:
+        genai_config["model"]["decoder"]["session_options"]["provider_options"] = []
+    with genai_config_path.open("w") as f:
+        json.dump(genai_config, f, indent=4)
+
+    build_text_embedding(args, input_model_path)
+
+    # Generate Olive configuration file for specific target
+    print("\nGenerating Olive configuration file...")
+    vision_config = generate_vision_config(args, input_model_path)
+    vision_output = olive_run(vision_config)
+    output_node = next(iter(vision_output.values())).get_top_ranked_nodes(1)[0]
+    model_path = output_node.model_config["config"]["model_path"]
+    pass_config = output_node.pass_run_config
+    resave_onnx_model(model_path, output_dir / "phi-3-v-128k-instruct-vision.onnx", pass_config)
+
+    text_config = generate_text_config(args, input_model_path)
+    try:
+        text_output = olive_run(text_config)
+        output_node = next(iter(text_output.values())).get_top_ranked_nodes(1)[0]
+        model_path = output_node.model_config["config"]["model_path"]
+        pass_config = output_node.pass_run_config
+        resave_onnx_model(model_path, output_dir / "phi-3-v-128k-instruct-text.onnx", pass_config)
+    except Exception:
+        config_link = "https://huggingface.co/microsoft/Phi-3-vision-128k-instruct-onnx-cpu/tree/main/cpu-int4-rtn-block-32-acc-level-4."
+        print(
+            "Ignore the error during Olive run.\nThis script will copy the "
+            "genai_config.json and processor_config.json from ",
+            config_link,
+        )
+        # even if the Olive run fails, partially generated model
+        # are still useful. Copy the model file to the output directory
+        text_config = RunConfig.parse_obj(text_config)
+        cache_dir = text_config.engine.cache_dir
+        pass_config = text_config.passes["builder"].config
+        # find the model file in the cache directory
+        model_path = list(cache_dir.rglob("*.onnx"))[-1]
+        resave_onnx_model(model_path, output_dir / "phi-3-v-128k-instruct-text.onnx", pass_config)
+    print("Model generation completed, output saved to ", output_dir)
+
+    if args.inference:
+        generate(output_dir)
+
+
+def generate_vision_config(args, input_model_path):
+    config = json.load((config_path / "vision_config.json").open())
+    config["input_model"]["model_path"] = input_model_path
+    config["input_model"]["model_script"] = config_path.parent / "scripts" / "user_script.py"
+
+    if args.precision == "fp16" or (args.precision == "int4" and args.target == "cuda"):
+        config["passes"]["convert"]["torch_dtype"] = "float16"
+    else:
+        config["passes"]["convert"]["torch_dtype"] = "float32"
+
+    if args.target == "cpu":
+        config["passes"]["convert"]["torch_dtype"] = "float32"
```

Comment:
nit: this looks redundant? Since the float16 case should only happen for cpu if we enforce the check earlier.